JASP-Engine/JASP/R/regressionlinearbayesian.R

```diff
@@ -611,1138 +384,1065 @@ RegressionLinearBayesian <- function (
-            width = w,
-            height = h,
-            convertible = TRUE,
-            status = "complete",
-            obj = content[["obj"]],
-            data = content[["png"]]
-        )
+.basregPlotInclusionProbabilities <- function(basregModel, jaspResults, options, status) {
+  if (!is.null(jaspResults[["inclusionProbabilitiesPlot"]]) ||
+      !options$plotInclusionProbabilities)
+    return()
+  
+  inclusionProbabilitiesPlot <- createJaspPlot(title = "Inclusion Probabilities", width = 700, height = 400)
+  jaspResults[["inclusionProbabilitiesPlot"]] <- inclusionProbabilitiesPlot
+  
+  inclusionProbabilitiesPlot$copyDependenciesFromJaspObject(jaspResults[["stateModel"]])
+  inclusionProbabilitiesPlot$dependOnOptions("plotInclusionProbabilities")
+  
+  if (!status$ready) {
+    return()
+  } else if (status$error) {
+    inclusionProbabilitiesPlot$error <- "badData"
+    return()
+  }
+  
+  .basregFillPlotInclusionProbabilities(basregModel, jaspResults, options)
+  
+  return()
+}
 
-    } else {
-        plot <- .makeEmptyPlot.basReg(title = title, status = status)
-    }
+.basregFillPlotInclusionProbabilities <- function(basregModel, jaspResults, options) {
+  
+  p <- try({
+    
+    probne0 = basregModel$probne0[-1]
+    variables = basregModel$namesx[-1] # 1:basregModel$n.vars
+    priorProb <- basregModel$priorprobsPredictor[1:basregModel$n.vars][-1]
+    
+    # reorder from low to high
+    o <- order(probne0, decreasing = FALSE)
+    probne0 <- probne0[o]
+    variables <- variables[o]
+    priorProb <- priorProb[o]
+    
+    width <- .8 # width of the bars
+    dfBar <- data.frame(
+      x = factor(variables, levels = variables),
+      y = probne0
+    )
+    dfLine <- data.frame(
+      x = rep(1:(basregModel$n.vars-1), each = 2) + c(-width/2, width/2),
+      y = rep(priorProb, each = 2),
+      g = rep(factor(variables), each = 2),
+      g0 = factor(1)
+    )
+    base <- .1
+    yLimits <- c(0, base * ceiling(max(c(priorProb, probne0)) / base))
+    yBreaks <- seq(yLimits[1], yLimits[2], length.out = 5)
+    
+    g <- JASPgraphs::drawBars(dat = dfBar, width = width)
+    g <- JASPgraphs::drawLines(g, dat = dfLine,
+                               mapping = ggplot2::aes(x = x, y = y, group = g, linetype = g0), show.legend = TRUE) +
+      ggplot2::scale_y_continuous("Marginal Inclusion Probability", breaks = yBreaks, limits = yLimits) +
+      ggplot2::xlab("") +
+      ggplot2::scale_linetype_manual(name = "", values = 2, labels = "Prior\nInclusion\nProbabilities")
     
-    return (plot)
+    JASPgraphs::themeJasp(g, horizontal = TRUE, legend.position = "right") +
+      ggplot2::theme(
+        legend.title = ggplot2::element_text(size = .8*JASPgraphs::graphOptions("fontsize"))
+      )
     
+  })
+  
+  if (isTryError(p)) {
+    errorMessage <- paste("Plotting not possible:", .extractErrorMessage(p))
+    jaspResults[["inclusionProbabilitiesPlot"]]$error <- "badData" #FIXME: this should go
```

Comment:
In these (and other functions) why not pass the plotObject, rather than jaspResults? Should work exactly the same but is a bit more concise.

Reply:
Yeah, this is indeed much nicer